sql/log_event.h

```diff
@@ -2898,14 +2898,34 @@ class Rows_log_event : public virtual binary_log::Rows_event, public Log_event {
     Key_compare(KEY **ki = nullptr) : m_key_info(ki) {}
     bool operator()(uchar *k1, uchar *k2) const {
       return key_cmp2((*m_key_info)->key_part, k1, (*m_key_info)->key_length,
-                      k2, (*m_key_info)->key_length) < 0;
+                      k2, (*m_key_info)->key_length) == 0;
```

Comment:
Rename this helper class to `Key_equal`.
`Compare` may be misleading as it may mean `comparator` and in your scenario this is no longer the case. All you do now is checking for equality.